tests/linalg_test.py

```diff
@@ -1373,6 +1373,82 @@ def expm(x):
+        jtu.cases_from_list({
+            "testcase_name":
+            "_shape={}".format(jtu.format_shape_dtype_string(shape, dtype)),
+            "shape": shape, "dtype": dtype
+        } for shape in [(4, 4), (15, 15), (50, 50), (100, 100)]
+                            for dtype in float_types + complex_types))
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testSchur(self, shape, dtype):
+      rng = jtu.rand_default(self.rng())
+      args_maker = lambda: [rng(shape, dtype)]
+
+      self._CheckAgainstNumpy(osp.linalg.schur, jsp.linalg.schur, args_maker)
+      self._CompileAndCheck(jsp.linalg.schur, args_maker)
+
+  @parameterized.named_parameters(
+    jtu.cases_from_list({
+        "testcase_name":
+        "_shape={}".format(jtu.format_shape_dtype_string(shape, dtype)),
+        "shape" : shape, "dtype" : dtype
+    } for shape in [(4, 4), (15, 15), (50, 50), (100, 100)]
+      for dtype in float_types + complex_types))
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testSqrtmPSDMatrix(self, shape, dtype):
+    # Checks against scipy.linalg.sqrtm when the principal square root
+    # is guaranteed to be unique (i.e no negative real eigenvalue)
+    rng = jtu.rand_default(self.rng())
+    arg = rng(shape, dtype)
+    mat = arg @ arg.T
+    args_maker = lambda : [mat]
+    if dtype == np.float32 or dtype == np.complex64:
+        tol = 1e-4
+    else:
+        tol = 1e-8
+    self._CheckAgainstNumpy(osp.linalg.sqrtm,
+                            jsp.linalg.sqrtm,
+                            args_maker,
+                            tol=tol,
+                            check_dtypes=False)
+    self._CompileAndCheck(jsp.linalg.sqrtm, args_maker)
+
+  @parameterized.named_parameters(
+  jtu.cases_from_list({
+      "testcase_name":
+      "_shape={}".format(jtu.format_shape_dtype_string(shape, dtype)),
+      "shape" : shape, "dtype" : dtype
+  } for shape in [(4, 4), (15, 15), (50, 50), (100, 100)]
+    for dtype in float_types + complex_types))
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testSqrtmGenMatrix(self, shape, dtype):
+    rng = jtu.rand_default(self.rng())
+    arg = rng(shape, dtype)
+    if dtype == np.float32 or dtype == np.complex64:
+      tol = 1e-3
+    else:
+      tol = 1e-8
+    R = jsp.linalg.sqrtm(arg)
+    self.assertAllClose(R @ R, arg, atol=tol, check_dtypes=False)
+
+  @parameterized.named_parameters(
+  jtu.cases_from_list({
+      "testcase_name":
+      "_diag={}".format((diag, dtype)),
+      "diag" : diag, "dtype" : dtype
+  } for diag in [[1, 0, 0], [0, 4, 0], [0, 0, 0, 9], [0, 0, 9, 0, 0, 4]]
+    for dtype in float_types + complex_types))
+  @jtu.skip_on_devices("gpu", "tpu")
+  def testSqrtmEdgeCase(self, diag, dtype):
+    """
+    Tests the zero numerator condition
+    """
+    mat = jnp.diag(jnp.array(diag)).astype(dtype)
+    root = jsp.linalg.sqrtm(mat)
+
+    self.assertAllClose(jnp.matmul(root, root, precision=lax.Precision.HIGHEST),
+                        mat, atol=1e-3, check_dtypes=False)
```

Comment:
Why not directly test for the expected solution here (e.g., against the elementwise `jnp.sqrt(mat)`), rather than relying on squaring `root`? That seems a little more rigorous to me, unless `sqrtm` has some ambiguity about which root it returns.